pkgs/development/compilers/dmd/default.nix

```diff
@@ -10,9 +10,15 @@ stdenv.mkDerivation {
 
   buildInputs = [ unzip curl makeWrapper ];
 
-  # Allow to use "clang++", commented in Makefile
   postPatch = stdenv.lib.optionalString stdenv.isDarwin ''
-      substituteInPlace src/dmd/posix.mak --replace g++ clang++
+      # Allow to use "clang++", commented in Makefile
+      substituteInPlace src/dmd/posix.mak \
+          --replace g++ clang++ \
+          --replace MACOSX_DEPLOYMENT_TARGET MACOSX_DEPLOYMENT_TARGET_
+
+      # Was not able to compile on darwin due to "__inline_isnanl"
+      # being undefined.
+      substituteInPlace src/dmd/root/port.c --replace __inline_isnanl __inline_isnan
```

Comment:
I am not too confident about this change, but did not manage to compile it at all with the most recent commit of nixpkgs.